pkg/controllers/workgenerator/controller.go

```diff
@@ -388,11 +464,49 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 					return
 				}
 				// Make sure the work is not deleted behind our back
+				klog.V(2).InfoS("Received a work delete event", "work", klog.KObj(evt.Object), "parentBindingName", parentBindingName)
+				queue.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+					Name: parentBindingName,
+				}})
+			},
+			// we care about work update event as we want to know when a work is applied so that we can
+			// update the corresponding resource binding status fast.
+			UpdateFunc: func(evt event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+				if evt.ObjectOld == nil || evt.ObjectNew == nil {
+					klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("updateEvent %v received with no matadata", evt)),
+						"Failed to process an update event for work object")
+					return
+				}
+				parentBindingName, exist := evt.ObjectNew.GetLabels()[fleetv1beta1.ParentBindingLabel]
+				if !exist {
+					klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("work has no binding parent")),
+						"Could not find the parent binding label", "updated work", evt.ObjectNew, "existing label", evt.ObjectNew.GetLabels())
+					return
+				}
+				oldWork, ok := evt.ObjectOld.(*workv1alpha1.Work)
+				if !ok {
+					klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("received old object %v not a work object", evt.ObjectOld)),
+						"Failed to process an update event for work object")
+					return
+				}
+				newWork, ok := evt.ObjectNew.(*workv1alpha1.Work)
+				if !ok {
+					klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("received new object %v not a work object", evt.ObjectNew)),
+						"Failed to process an update event for work object")
+					return
+				}
+				oldAppliedStatus := meta.FindStatusCondition(oldWork.Status.Conditions, workapi.ConditionTypeApplied)
+				newAppliedStatus := meta.FindStatusCondition(newWork.Status.Conditions, workapi.ConditionTypeApplied)
+				if condition.IsConditionStatusTrue(oldAppliedStatus, oldWork.GetGeneration()) == condition.IsConditionStatusTrue(newAppliedStatus, newWork.GetGeneration()) {
+					klog.V(4).InfoS("Applied condition in the work didn't change", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
```

Comment:
add a comment to indicate that this controller only cares about if work condition is applied or not.

work condition can change from unknown to false, but it won't change the binding applied status.